src/Simplify_Call.cpp

```diff
@@ -40,7 +40,9 @@ Expr Simplify::visit(const Call *op, ExprInfo *bounds) {
             }
 
             if (ib >= 0) {
-                if (ib < std::min(t.bits(), 64) - 1) {
+                int max_shift = std::min(t.bits(), 64);
```

Comment:
What is this line doing? It seems it should either be an assert or not limit the size. (128-bit integers are a thing on the cutting edge of computer architecture.) I assume the context has already ensured that t is a scalar, not a vector type.

Reply:
Looks to me like it's verifying that the 64-bit constant we produce (to hand off to Mul or Div) won't overflow an int64. That said, now that you point this out, the logic I'm adding would in fact overflow if the type is uint64. Maybe this whole clause would be clearer if split into completely separate int-vs-uint sections.